File: Astronomy_ObjectiveC/Astronomy_ObjectiveC/Controller/View Controllers/ViewController.m
//
//  ViewController.m
//  Astronomy_ObjectiveC
//
//  Created by Kenny on 6/15/20.
//  Copyright © 2020 Hazy Studios. All rights reserved.
//

#import "ViewController.h"
#import "KMLRoverController.h"
#import "KMLManifest.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    KMLRoverController *roverController = [[KMLRoverController alloc] init];
    [roverController fetchSolsFromRoverWithName:@"Curiosity" completion:^(KMLManifest *manifest) {
        NSLog(@"%@", manifest.sols);
    }];
}


@end

File: CARAstronomy/CARAstronomy/Controller/View Controllers/ViewController.swift
//
//  ViewController.swift
//  CARAstronomy
//
//  Created by Chad Rutherford on 2/24/20.
//  Copyright © 2020 Chad Rutherford. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let manifestFetcher = CARManifestFetcher()
    var manifest: CARMarsMissionManifest?
    let solFetcher = CARSolFetcher()

    override func viewDidLoad() {
        super.viewDidLoad()
        manifestFetcher.fetchManifest(forRover: "curiosity") { manifest, error in
            guard error == nil else { return }
            
            guard let manifest = manifest else { return }
            self.manifest = manifest
        }
        
        solFetcher.fetchPhotos(forRover: "curiosity", withSol: 1000) { sols, error in
            guard error == nil else { return }
            guard let sols = sols else { return }
            print(sols)
        }
    }
}
